### PR TITLE
Conservative actions

### PR DESCRIPTION
**Pull request by wthrowe**
A particular thing to consider here is the handling of source terms.  I currently have the system specifying which variables have sources and then only passing those to the source calculation functions (and in a later PR to the RHS).

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- Follows [code review guidelines](https://sxs-collaboration.github.io/spectre/code_review_guide.html)
- Code has documentation and unit tests
- Private member variables have a trailing underscore
- Do not use [Hungarian notation](https://en.wikipedia.org/wiki/Hungarian_notation), e.g. `double* pd_blah` is bad
- Header order:
  1. hpp corresponding to cpp (only in cpp files) or `tests/Unit/TestingFramework.hpp` (only in tests)
  2. Blank line (only in cpp files and tests)
  3. STL and externals (in alphabetical order)
  4. Blank line
  5. SpECTRE includes (in alphabetical order)
- File lists in CMake are alphabetical
- Correct `noexcept` specification for functions (if unsure, mark `noexcept`)
- Mark objects `const` whenever possible
- Almost always `auto`, except with expression templates, i.e. `DataVector`
- All commits for performance changes provide quantitative evidence and the tests used to obtain said evidence.
- Make sure error messages are helpful, e.g. "The number of grid points in the matrix 'F' is not the same as the number of grid points in the determinant."
- Prefix commits addressing PR requests with `fixup`. These commits are flagged
  by Travis so we do not accidentally merge them.
- Include what you use, prefer forward declarations in hpp files.
- Explicitly make numbers floating point, e.g. `2.` or `2.0` over `2`
- Use `Tensor`'s non-member get if the indices are constant expressions, e.g.
  `get<1>(tensor)`

### CODE REVIEW

**Review by osheamonn — DISMISSED** [review]

**Comment by kidder** [comment]
test name function

**Review by kidder — COMMENTED** [review]
@nilsdeppe please look at this

**Review by nilsdeppe — DISMISSED** [review]

File: src/DataStructures/DataBox/Prefixes.hpp [hunk 1]
```diff
@@ -58,6 +58,15 @@ struct Flux<Tag, VolumeDim, Fr,
 };
 /// \endcond
 
+/// \ingroup DataBoxTagsGroup
+/// \brief Prefix indicating a source term
+template <typename Tag>
+struct Source : db::PrefixTag, db::SimpleTag {
+  using type = db::item_type<Tag>;
+  using tag = Tag;
+  static std::string name() noexcept { return "Source(" + Tag::name() + ")"; }
```
Comment (nilsdeppe) [inline src/DataStructures/DataBox/Prefixes.hpp:69]: Could you test the name works as expected?

**Review by kidder — APPROVED** [review]
@nilsdeppe please look at

**Review by mar-celine — APPROVED** [review]

**Pull request merged**